lib/src/Uncertainty/Distribution/ClaytonCopula.cxx

```diff
@@ -385,14 +385,19 @@ Point ClaytonCopula::computeQuantile(const Scalar prob,
                                      const Bool tail) const
 {
   if (!((prob >= 0.0) && (prob <= 1.0))) throw InvalidArgumentException(HERE) << "Error: cannot compute a quantile for a probability level outside of [0, 1]";
-  const Scalar q = tail ? 1.0 - prob : prob;
-  // Special case for boarding values
-  if (q == 0.0) return getRange().getLowerBound();
-  if (q == 1.0) return getRange().getUpperBound();
+  // Special case for boundary values
+  if (prob == 0.0) return tail ? getRange().getUpperBound() : getRange().getLowerBound();
+  if (prob == 1.0) return tail ? getRange().getLowerBound() : getRange().getUpperBound();
   // Independent case
-  if (theta_ == 0.0) return Point(2, std::sqrt(q));
+  if (theta_ == 0.0) return Point(2, std::sqrt(tail ? 1.0 - prob : prob));
   // General case
-  return Point(2, std::exp((M_LN2 - log1p(std::pow(q, -theta_))) / theta_));
+  if (tail)
+  {
+    if (prob < std::sqrt(SpecFunc::ScalarEpsilon)) return Point(2, 1.0 - prob / 2.0);
```

Comment:
The code seems structurally correct from my point of view. But it introduces 4 different ways to return a value. Hence, this particular commit should introduce 4 different unit tests which individually check that each branch produce the correct value. Otherwise, a new bug might be introduced by this PR without being noticed by the current machinery. The initial problem was a slight numerical difference on different machines, which is much less a significant problem in general.

Reply:
I agree. Done.